test/integration/tas/tas_job_test.go

```diff
@@ -0,0 +1,602 @@
+									Count: 1,
+									Values: []string{
+										"b1",
+										"r1",
+									},
+								},
+								{
+									Count: 1,
+									Values: []string{
+										"b1",
+										"r2",
+									},
+								},
+							},
+						},
+					))
+				})
+
+				ginkgo.By("creating the second workload to see it lands on another rack", func() {
+					wl2 = testing.MakeWorkload("wl2", ns.Name).
+						Queue(localQueue.Name).Request(corev1.ResourceCPU, "1").Obj()
+					wl2.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Required: ptr.To(tasRackLabel),
+					}
+					gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
+				})
+
+				ginkgo.By("verify the workload is admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1, wl2)
+					util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 2)
+				})
+
+				ginkgo.By("verify admission for the workload2", func() {
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl2), wl2)).To(gomega.Succeed())
+					gomega.Expect(wl2.Status.Admission.PodSetAssignments[0].TopologyAssignment).Should(gomega.BeComparableTo(
+						&kueue.TopologyAssignment{
+							Levels: []string{
+								tasBlockLabel,
+								tasRackLabel,
+							},
+							Domains: []kueue.TopologyDomainAssignment{
+								{
+									Count: 1,
+									Values: []string{
+										"b2",
+										"r1",
+									},
+								},
+							},
+						},
+					))
+				})
+
+				ginkgo.By("creating the wl3", func() {
+					wl3 = testing.MakeWorkload("wl3", ns.Name).
+						Queue(localQueue.Name).Request(corev1.ResourceCPU, "1").Obj()
+					wl3.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Required: ptr.To(tasRackLabel),
+					}
+					gomega.Expect(k8sClient.Create(ctx, wl3)).Should(gomega.Succeed())
+				})
+
+				ginkgo.By("verify the wl3 is admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1, wl2, wl3)
+					util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 3)
+				})
+
+				ginkgo.By("verify admission for the wl3", func() {
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl3), wl3)).To(gomega.Succeed())
+					gomega.Expect(wl3.Status.Admission.PodSetAssignments[0].TopologyAssignment).Should(gomega.BeComparableTo(
+						&kueue.TopologyAssignment{
+							Levels: []string{
+								tasBlockLabel,
+								tasRackLabel,
+							},
+							Domains: []kueue.TopologyDomainAssignment{
+								{
+									Count: 1,
+									Values: []string{
+										"b2",
+										"r2",
+									},
+								},
+							},
+						},
+					))
+				})
+
+				ginkgo.By("creating wl4", func() {
+					wl4 = testing.MakeWorkload("wl4", ns.Name).
+						Queue(localQueue.Name).Request(corev1.ResourceCPU, "1").Obj()
+					wl4.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Required: ptr.To(tasRackLabel),
+					}
+					gomega.Expect(k8sClient.Create(ctx, wl4)).Should(gomega.Succeed())
+				})
+
```

Comment:
Can we check if wl4 is not admitted before wl3 is finished?

Reply:
sure, added a check: util.ExpectWorkloadsToBePending(ctx, k8sClient, wl4)